pkgs/by-name/li/librum/package.nix

```diff
@@ -0,0 +1,65 @@
+{ lib
+, mupdf
+, stdenv
+, fetchFromGitHub
+, substituteAll
+, cmake
+, qt6
+}:
+
+let
+  mupdf-cxx = mupdf.override { enableCxx = true; };
+in
+stdenv.mkDerivation rec {
+  pname = "librum";
+  version = "0.12.1";
+
+  src = fetchFromGitHub {
+    owner = "Librum-Reader";
+    repo = "Librum";
+    rev = "v.${version}";
+    fetchSubmodules = true;
+    hash = "sha256-/QxTWlTMoXykPe3z+mmn6eaGRJDu2IX8BJPcXi1gUqQ=";
+  };
+
+  patches = [
+    (substituteAll {
+      src = ./use_mupdf_in_nixpkgs.patch;
+      nixMupdfLibPath = "${mupdf-cxx.out}/lib";
+      nixMupdfIncludePath = "${mupdf-cxx.dev}/include";
+    })
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    qt6.qttools
+    qt6.wrapQtAppsHook
+  ];
```

Comment:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [ desktopToDarwinBundle ];
```

Reply:
Wait, Darwin doesn’t recognize .desktop files?

Reply:
> Wait, Darwin doesn’t recognize .desktop files?

No)

Reply:
Result is:
```
$ tree result
result
├── Applications
│   └── Librum.app
│       └── Contents
│           ├── Info.plist
│           ├── MacOS
│           │   └── Librum -> ../../../../bin/librum
│           └── Resources
│               └── librum.icns
├── bin
│   └── librum
├── lib
│   ├── libadapters.dylib
│   ├── libapplication.dylib
│   ├── libdomain.dylib
│   ├── libinfrastructure.dylib
│   └── libpresentation.dylib
└── share
    ├── applications
    │   └── librum.desktop
    └── pixmaps
        └── librum.svg
```

Reply:
Is this the only way this application will appear on the launcher page? (If so, is nixpkgs missing tons of `desktopToDarwinBundle`)

Reply:
Many Qt-based apps produce proper *.app bundle as build result, so no need `desktopToDarwinBundle`. But yes, there is still many GUI apps where `desktopToDarwinBundle` would be good to add.